### ISSUE TITLE
Perl_no_op() warnings call Perl_warner() more than once, which interferes with warnings fatal and __WARN__ hooks

### ISSUE THREAD

**Issue by philiprbrenan** [Perl/perl5#20425]
Without warnings enabled we get a helpful message telling us the name of the missing bareword such as  `Do you need to predeclare id_deeply?` in the following:
```
#!/usr/bin/perl
use v5.30;
sub pick{}

id_deeply pick;

# Bareword found where operator expected at ... line 5, near "id_deeply pick"
# 	(Do you need to predeclare id_deeply?)
# syntax error at ... line 5, near "id_deeply pick"
```

But with `warnings FATAL` enabled this helpful message is omitted. 

```
#!/usr/bin/perl
use v5.30;
use warnings FATAL => qw(all);
sub pick{}

id_deeply pick;

# Bareword found where operator expected at ... line 6, near "id_deeply pick"
```

Please could you provide this useful information even when warnings are enabled?  Doing so would save programmer time when it is not immediately obvious which of the two words is the one causing the problem.

**Comment by demerphq** [comment]
Hi thanks for the report. I agree with you, this is improper behavior I have pushed a patch to fix this: #20427. I have also done an audit for other cases, and I didn't spot any of the same form (eg adding warning data in parenthesized hints after the main warning message). This is problematic not just for fatal warnings but also for `$SIG{__WARN__}` which gets called twice, with parts of the overall message, from what is supposed to be a single warning.

**Comment by demerphq** [comment]
thanks, this has been fixed in blead and will be available in the next release of perl.
